### PR TITLE
Basic CRD scaffold

### PR DESCRIPTION
**Pull request by ccremer**
Sets up Kubebuilder scaffolding to generate empty resource.
Actual spec fields is done later in another PR

### CODE REVIEW

**Comment by ccremer** [comment]
Note: Currently builds upon #1 , but should be rebased later

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+builds:
+- binary: control-api
+  env:
+    - CGO_ENABLED=0 # this is needed otherwise the Docker image build is faulty
+  goarch:
+    - amd64
```
Comment (bastjan) [inline .goreleaser.yml:6]: Maybe we could add `arm64` too? 😊

**Review by bastjan — APPROVED** [review]
LGTM

To be completely honest i don't like `goreleaser` in a "container first" project. It makes builds intransparent since it does not support building everything in a container. I prefer a Dockerfile that works without special prerequisites and `buildx` actions. But I feel like that should be a team discussion...

File: .goreleaser.yml (hunk 1 above)
Comment (ccremer) [reply .goreleaser.yml:6]: Why? Any controller or API server we build will run exclusively on OpenShift on x86

**Comment by ccremer** [comment]
> To be completely honest i don't like `goreleaser` in a "container first" project. It makes builds intransparent since it does not support building everything in a container. I prefer a Dockerfile that works without special prerequisites and `buildx` actions. But I feel like that should be a team discussion...

I would understand your point if the programming language is something different than our main language Go, e.g. Java (like we did in Keycloak extensions). However, with Go as our primary language we all have a Go setup, and that's basically all that is required besides `make` and `docker`.

Besides, Goreleaser would support building everything in a container, one doesn't need to build the binary first and copy that into the image. But then in GH actions we couldn't make sensible use of Go's package caching.

File: .goreleaser.yml (hunk 1 above)
Comment (bastjan) [reply .goreleaser.yml]: Is there never a need for a local setup?

Comment (ccremer) [reply .goreleaser.yml]: It is, but not with Goreleaser. Goreleaser is meant purely for releasing in GH actions.
I guess I'll need to re-add GOOS and GOARCH to Makefile.

Comment (ccremer) [reply .goreleaser.yml]: I've added instructions in README how to build with ARM/mac

**Pull request merged**